core/src/player.rs

```diff
@@ -1184,12 +1187,25 @@ impl Player {
             let needs_render = if events.is_empty() {
                 false
             } else {
+                let mut refresh = false;
                 for (object, event) in events {
-                    if !object.as_displayobject().removed() {
+                    let display_object = object.as_displayobject();
+                    if !display_object.removed() {
                         object.handle_clip_event(context, event);
                     }
+                    if !refresh && event.is_button_event() {
```

Comment:
Good idea on avoiding the redeundant work if `refresh` is already set here, because `is_button_event` requires an call into AVM and is indeed nice to avoid this when we can :-)